### PR TITLE
test_assertion_location

### PR DESCRIPTION
**Pull request by blueyed**
Ref: https://github.com/pytest-dev/pytest/issues/5754

### CODE REVIEW

**Comment by blueyed** [comment]
Failed as expected under coverage: https://travis-ci.org/pytest-dev/pytest/jobs/573192533#L786

/cc @nedbat @asottile 
Any idea what is happening here?

File: testing/test_assertrewrite.py [hunk 1]
```diff
@@ -1580,3 +1580,21 @@ def fake_mkdir(p, *, exc):
     with pytest.raises(OSError) as exc_info:
         try_mkdir(str(p))
     assert exc_info.value.errno == errno.ECHILD
+
+
+def test_assertion_location(testdir):
```
Comment (blueyed) [inline testing/test_assertrewrite.py]: Should/could be included with an existing basic test probably.

Comment (blueyed) [reply testing/test_assertrewrite.py]: Could not find a better place.. :/

**Comment by asottile** [comment]
since it's global, all `assert`s will share those ast instances (with last-one-wins positioning information) -- not sure why coverage changes it though 🤔 

patch looks good though 👍

**Comment by blueyed** [comment]
@asottile 
Thanks!

@nedbat 
Still interested in your input - might also indicate some bug / issue with Coverage.py after all.

**Pull request merged**